src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/InterceptedProjectProperties.cs

```diff
@@ -98,5 +110,34 @@ public void SetRuleContext(Rule rule)
             var ruleAwareProperties = DelegatedProperties as IRuleAwareProjectProperties;
             ruleAwareProperties?.SetRuleContext(rule);
         }
+
+        private IInterceptingPropertyValueProvider? GetFilteredProvider(string propertyName, (List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> Providers, bool Filtered) valueTuple)
```

Comment:
Throughout this code in this class, the type `(List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> Providers, bool Filtered)` is used heavily, and makes the code quite hard to read.

Consider promoting the value tuple to a dedicated nested type, and moving the `GetFilteredProvider` method to that new type.

```c#
private readonly struct Providers
{
    public List<Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>> Exports { get; }

    public bool Filtered { get; }

    public IInterceptingPropertyValueProvider? GetFilteredProvider(string propertyName) { ... }
}
```